javascript/code-challenges/challenges-01.test.js

```diff
@@ -0,0 +1,222 @@
+
+  arr.forEach((string) => tallArray.push(string.toUpperCase()));
+
+  return tallArray;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function named `greeting` that takes in a single string and returns the string in all uppercase letters, and followed by an "!".
+
+Then, write a function named `speaker` that takes in an array of strings and a callback function. 
+
+Use `forEach` to build a new array of strings, each string modified by the callback. Return the new array. 
+------------------------------------------------------------------------------------------------ */
+
+const greeting = (word) => {
+  return word.toUpperCase() + '!';
+};
+
+const speaker = (words, callback) => {
+  const callBackArray = [];
+
+  words.forEach((word) => callBackArray.push(callback(word)));
+
+  return callBackArray;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 5
+
+Write a function named addValues that takes in an array and a value and pushes the value into the array. This function does not need a return statement.
+
+Then, write a function named addNumbers that takes in four arguments:
+  - A number to be added to an array
+  - An array into which the number should be added
+  - The number of times the number should be added
+  - A callback function to use to add the numbers to the array (Hint: you already defined it)
+
+Within the addNumbers function, invoke the callback function as many times as necessary, based on the third argument of the addNumbers function.
+
+Return the modified array.
+------------------------------------------------------------------------------------------------ */
+
+const addValues = (arr, value) => {
+  arr.push(value);
+};
+
+const addNumbers = (num, arr, times, callback) => {
+  const addedArray = arr;
+
+  for (let i = 0; i < times; i++) {
+    callback(addedArray, num);
+  }
+
+  return addedArray;
+};
+
+/* ------------------------------------------------------------------------------------------------
+
+CHALLENGE 6
+
+Write a function named createList that takes in an array of the current store intentory.
+
+The inventory is formatted like this:
+[
+  { name: 'apples', available: true },
+  { name: 'pears', available: true },
+  { name: 'oranges', available: false },
+  { name: 'bananas', available: true },
+  { name: 'blueberries', available: false }
+]
+
+This function should use forEach to populate your grocery list based on the store's inventory. If the item is available, add it to your list. Return the final list.
+------------------------------------------------------------------------------------------------ */
+
+const createList = (availableItems) => {
+  const availableArray = [];
+
+  availableItems.forEach( (item) => {
+
+    if (item['available']) {
+      availableArray.push(item['name']);
+    }
+  });
+  return availableArray;
+};
```

Comment:
Really lovely work! 💯 

Feedback here regarding variable names: I love how descriptive your names are: `excitedArr`,` availableArr`, `tallArr` etc. but I bet you could go further with this. 

I would consider: What does each one represent? What are the return values?
For example: for `availableArr`, it holds the list items that you are returning. So a more descriptive name might be `availableItemsList` or `availableItemsArr`. 

For `excitedArr`, it isn't quite clear what the arr holds. We can guess what it holds based off the challenge description, but what if the description for the challenge weren't there? Would we know?

Just some food for thought for moving forward. Great work again here! 👍